### ISSUE TITLE
Bracket in comment increases indent level

### ISSUE THREAD

**Issue by MatthiasPortzel** [ziglang/sublime-zig-language#72]
Press Enter at the end of the line below.
```zig
// hello {
```
The indent level increases when it should not. Other sublime syntaxes do not have this behavior.

**Comment by aidenfoxivey** [comment]
> Press Enter at the end of the line below.
> 
> ```zig
> // hello {
> ```
> 
> The indent level increases when it should not. Other sublime syntaxes do not have this behavior.

Is this still the case? I installed the latest version and didn't see it - but maybe there's something else going on here.

**Comment by sigod** [comment]
Yes, this happens on my side. But it also happens with Bash, D, JSON, Lua, R, Ruby, and Rust, but not with C, C#, C++, Java, JavaScript, TypeScript...

**Comment by aidenfoxivey** [comment]
![CleanShot 2024-08-10 at 18 04 43](https://github.com/user-attachments/assets/df715c8b-059d-48e9-8e4f-a25b3402ad68)
Is this the same or different behaviour as you're having?

**Comment by sigod** [comment]
No. Here's a gif of what I see:
![zig-indent-level-72](https://github.com/user-attachments/assets/c7bfa473-9096-44c3-8ed6-0ae6fac98843)

**Comment by aidenfoxivey** [comment]
Very strange, hmm.

**Comment by aidenfoxivey** [comment]
> No. Here's a gif of what I see: ![zig-indent-level-72](https://private-user-images.githubusercontent.com/1175750/356840607-c7bfa473-9096-44c3-8ed6-0ae6fac98843.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjMzOTMyMTEsIm5iZiI6MTcyMzM5MjkxMSwicGF0aCI6Ii8xMTc1NzUwLzM1Njg0MDYwNy1jN2JmYTQ3My05MDk2LTQ0YzMtOGVkNi0wYWU2ZmFjOTg4NDMuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgxMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MTFUMTYxNTExWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YjM1Mjc0NjIyNzNmZjIzZWRjYzc3ZWJjNzZjY2IwM2RlMjVmMDJjY2QzMjM2ZTlkMDNjOWM1ZjM2OWNiZDVmNyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.D02wLuuBcSR4LsnpGucndc9hzdxwfs9Cp1T8jcKELPk) [ ![zig-indent-level-72](https://private-user-images.githubusercontent.com/1175750/356840607-c7bfa473-9096-44c3-8ed6-0ae6fac98843.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjMzOTMyMTEsIm5iZiI6MTcyMzM5MjkxMSwicGF0aCI6Ii8xMTc1NzUwLzM1Njg0MDYwNy1jN2JmYTQ3My05MDk2LTQ0YzMtOGVkNi0wYWU2ZmFjOTg4NDMuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgxMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MTFUMTYxNTExWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YjM1Mjc0NjIyNzNmZjIzZWRjYzc3ZWJjNzZjY2IwM2RlMjVmMDJjY2QzMjM2ZTlkMDNjOWM1ZjM2OWNiZDVmNyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.D02wLuuBcSR4LsnpGucndc9hzdxwfs9Cp1T8jcKELPk) ](https://private-user-images.githubusercontent.com/1175750/356840607-c7bfa473-9096-44c3-8ed6-0ae6fac98843.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjMzOTMyMTEsIm5iZiI6MTcyMzM5MjkxMSwicGF0aCI6Ii8xMTc1NzUwLzM1Njg0MDYwNy1jN2JmYTQ3My05MDk2LTQ0YzMtOGVkNi0wYWU2ZmFjOTg4NDMuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgxMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MTFUMTYxNTExWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YjM1Mjc0NjIyNzNmZjIzZWRjYzc3ZWJjNzZjY2IwM2RlMjVmMDJjY2QzMjM2ZTlkMDNjOWM1ZjM2OWNiZDVmNyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.D02wLuuBcSR4LsnpGucndc9hzdxwfs9Cp1T8jcKELPk) [ ](https://private-user-images.githubusercontent.com/1175750/356840607-c7bfa473-9096-44c3-8ed6-0ae6fac98843.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjMzOTMyMTEsIm5iZiI6MTcyMzM5MjkxMSwicGF0aCI6Ii8xMTc1NzUwLzM1Njg0MDYwNy1jN2JmYTQ3My05MDk2LTQ0YzMtOGVkNi0wYWU2ZmFjOTg4NDMuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgxMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MTFUMTYxNTExWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YjM1Mjc0NjIyNzNmZjIzZWRjYzc3ZWJjNzZjY2IwM2RlMjVmMDJjY2QzMjM2ZTlkMDNjOWM1ZjM2OWNiZDVmNyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.D02wLuuBcSR4LsnpGucndc9hzdxwfs9Cp1T8jcKELPk)

https://github.com/tvi/Sublime-ARM-Assembly/blob/master/Syntaxes/ARMComment.tmPreferences

Maybe it's this TM_COMMENT_DISABLE_INDENT you need?

**Comment by aidenfoxivey** [comment]
Wait no, doesn't seem to be it. my bad

**Comment by aidenfoxivey** [comment]
Sorry for the long gap! I found a way to solve it:

```
		<key>increaseIndentPattern</key>
		<string>^(?!\s*//).*\{[^}"']*$</string>
		<key>decreaseIndentPattern</key>
		<string>^\s*\}</string>
```

I did check rust-enhanced in the meantime though and found that it still also has this issue.
